iModelCore/GeomLibs/geom/test/PolyfaceTest/t_ClipPlaneSet.cpp

```diff
@@ -2361,3 +2361,193 @@ TEST(ClipPlaneSet, AlignmentByMatchedArrays)
 #endif
     Check::ClearGeometry("ClipPlaneSet.AlignmentByMatchedArrays");
     }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+TEST(ClipPlaneSet,ClipNoneBoundary)
+    {
+    // curve
+    bvector<DPoint3d> poly1 {{0,10,0},{-10,0,0},{0,-10,0},{10,0,0},{0,10,0}};
+    CurveVector::BoundaryType boundaryType = CurveVector::BoundaryType::BOUNDARY_TYPE_None;
+    CurveVectorPtr curve = CurveVector::CreateLinear(poly1, boundaryType);
+    // clipper
+    bvector<DPoint3d> poly2 {{-7,-7,0},{7,-7,0},{7,7,0},{-7,7,0},{-7,-7,0}};
+    ConvexClipPlaneSet convexClipPlaneSet;
+    convexClipPlaneSet.ReloadSweptConvexPolygon(poly2, DVec3d::From(0, 0, 1), 0);
+    ClipPlaneSet clipPlaneSet(convexClipPlaneSet);
+    // perform the clip
+    bvector<CurveVectorPtr> clippedRegion;
+    bool ret = clipPlaneSet.ClipCurveVector(*curve, clippedRegion);
+    Check::False(ret);
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+TEST(ClipPlaneSet,ClipOpenBoundary)
+    {
+    // curve
+    bvector<DPoint3d> poly1 {{0,10,0},{-10,0,0},{0,-10,0},{10,0,0},{0,10,0}};
+    Check::SaveTransformed(poly1);
+    CurveVector::BoundaryType boundaryType = CurveVector::BoundaryType::BOUNDARY_TYPE_Open;
+    CurveVectorPtr curve = CurveVector::CreateLinear(poly1, boundaryType);
+    // clipper
+    bvector<DPoint3d> poly2 {{-7,-7,0},{7,-7,0},{7,7,0},{-7,7,0},{-7,-7,0}};
+    Check::SaveTransformed(poly2);
+    ConvexClipPlaneSet convexClipPlaneSet;
+    convexClipPlaneSet.ReloadSweptConvexPolygon(poly2, DVec3d::From(0, 0, 1), 0);
+    ClipPlaneSet clipPlaneSet(convexClipPlaneSet);
+    // perform the clip
+    bvector<CurveVectorPtr> clippedRegion;
+    bool ret = clipPlaneSet.ClipCurveVector(*curve, clippedRegion);
+    Check::Shift(30,0,0);
+    Check::True(ret);
+    for (CurveVectorPtr clippedCurve : clippedRegion)
+        Check::SaveTransformed(*clippedCurve);
+    Check::ClearGeometry("ClipPlaneSet.ClipOpenBoundary");
+    }
+
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+TEST(ClipPlaneSet,ClipNotOpenBoundary)

```

Comment:
I would combine both "ClipNotOpenBoundary" and "ClipOpenBoundary" into one test so you can compare their output to add another sanity check. Because the only difference in the inputs is the boundaryType of the curve, we expect both returned bvectors to contain the same clipped curves in the same order, so you should be able to iterate these bvectors in tandem and check that `clipped0[i]->IsSameStructureAndGeometry(*clipped1[i])` returns true.